components/search/SearchArea.vue

```diff
@@ -0,0 +1,314 @@
+);
+
+const filters = ref<{ value: string; label: string }[]>([]);
+const isOpen = ref(false);
+const selectedFilter = ref<{ value: string; label: string } | null>(null);
+const searchQuery = ref<string>("");
+const showSuggestions = ref<boolean>(false);
+const shouldShowSuggestions = ref<boolean>(false); // New state for keeping suggestions open
+const suggestions = ref<string[]>([]);
+
+const debouncedFetchSuggestions = debounce(async (query: string) => {
+  if (query.trim().length === 0) {
+    suggestions.value = [];
+    return;
+  }
+
+  // Fetch data from Supabase
+  const { data, error } = await supabase
+    .from("research_papers")
+    .select("title")
+    .ilike("title", `%${query}%`); // ilike for case-insensitive partial match
+
+  if (error) {
+    console.error("Error fetching suggestions:", error);
+    suggestions.value = [];
+  } else {
+    const queryLower = query.toLowerCase().trim();
+
+    // Scoring and sorting data based on a more complex relevance system
+    const rankedSuggestions = data
+      .map((paper: { title: string }) => {
+        const titleLower = paper.title.toLowerCase().trim();
+        let score = 0;
+
+        // Exact match (whole title matches query)
+        if (titleLower === queryLower) {
+          score += 100;
+        } else {
+          // Query starts the title (e.g., 'Map', 'Mana')
+          if (titleLower.startsWith(queryLower)) {
+            score += 80;
+          }
+
+          // Query is a whole word at the start of the title
+          const words = titleLower.split(/\s+/); // split title into words
+          if (words[0] === queryLower) {
+            score += 70;
+          }
+
+          // Query is part of the title
+          if (titleLower.includes(queryLower)) {
+            score += 50;
+          }
+
+          // Check for word proximity (closer match gets a higher score)
+          const indexOfQuery = titleLower.indexOf(queryLower);
+          if (indexOfQuery !== -1) {
+            // Closer to the start, higher score
+            score += 30 / (indexOfQuery + 1); // inverse proportional boost
+          }
+
+          // Penalize for distance between query parts (split across words)
+          if (
+            titleLower.split(/\s+/).some((word) => word.includes(queryLower))
+          ) {
+            score += 20;
+          }
+
+          // Shorter titles are given a higher score
+          score += 10 / titleLower.length;
+        }
+
+        return { title: paper.title, score };
+      })
+      // Sort by score first (highest score at the top)
+      .sort((a, b) => b.score - a.score)
+      // Limit results to 5 suggestions
+      .slice(0, 5);
+
+    // Update suggestions with the ranked and sorted titles
+    suggestions.value = rankedSuggestions.map((s) => s.title);
+  }
+
+  shouldShowSuggestions.value = true;
+}, 300);
```

Comment:
No querying in ui layer, transfer it in services layer. Change the entire code.

Reply:
done